torch/utils/cpp_extension.py

```diff
@@ -423,6 +423,8 @@ def unix_wrap_ninja_compile(sources,
                                              include_dirs, sources,
                                              depends, extra_postargs)
             common_cflags = self.compiler._get_cc_args(pp_opts, debug, extra_preargs)
+            from shlex import quote
+            common_cflags = [quote(f) for f in common_cflags]
```

Comment:
Let's use `quote` inside of the `sanitize_flags` function instead of at this callsite.

https://github.com/pytorch/pytorch/blob/1ef77f90454a5cac154aa171ebb9a676c4d707b9/torch/utils/cpp_extension.py#L1627-L1637

That should also fix the spaces issue on windows (windows ninja build goes through a different path but all ninja paths ultimately calls sanitize_flags).

Reply:
I tried this originally, but it turns out that windows flags are already quoted by `_nt_quote_args` before.

Reply:
Oh, that's a great point. It looks like quoting might be [different on windows...](https://github.com/python/cpython/commit/e2b4452d0c3cc869d27e382100fda4466d39314b) :/  (shlex.quote produces single quotes on my linux machine, don't have a windows machine to test with).

To be consistent with the windows quoting, could we quote "cflags", "post_cflags", "cuda_cflags", and "cuda_post_cflags" in the unix branch?  It doesn't seem to hurt to quote ldflags as well https://github.com/pytorch/pytorch/blob/1ef77f90454a5cac154aa171ebb9a676c4d707b9/torch/utils/cpp_extension.py#L453-L462